packages/plugin-legacy/index.js

```diff
@@ -220,6 +225,10 @@ function viteLegacyPlugin(options = {}) {
 
     renderChunk(raw, chunk, opts) {
       if (!isLegacyOutput(opts)) {
+        if (genDynamicFallback && chunk.isEntry) {
+          raw = blankDynamicImport + raw
+        }
```

Comment:
We need to respect source maps here. You can use magic string for this, checkout that later in this function it is being used to modify the code while keeping track of the source maps changes, here
https://github.com/vitejs/vite/blob/da2a41de65670b5e766c596fa81443cf2d3fdf11/packages/plugin-legacy/index.js#L234

Maybe this logic could be moved there so only one magic string is used?

Reply:
Nice catch, I completely forgot about source maps
I've moved some code around to always use magic string for operations to avoid logic duplication